packages/ssx-core/src/server/types.ts

```diff
@@ -88,3 +91,85 @@ export enum SSXEventLogTypes {
   /** Event type definition. */
   // EVENT = "event",
 }
+
+/**
+ * SSX-Server is a server-side library made to work with the SSX client libraries.
+ * SSX-Server is the base class that takes in a configuration object and works
+ * with various middleware libraries to add authentication and metrics to your server.
+ */
+export abstract class SSXServerBaseClass extends EventEmitter {
```

Comment:
This was a refactor to support `ssx-server-middleware` relying on class definintions used in `ssx-server`. This change was to prevent circular dependencies. However, I think we should move our base class definintions to `ssx-core` and have implementations in the respective projects in order to improve extensibility

Reply:
agree!